### PR TITLE
Bump the DNS library and revendor

### LINKED ISSUE #2493: Please update "github.com/miekg/dns"
[github.com/miekg/dns](https://github.com/miekg/dns/releases) v1.1.26 fixes CVE-2019-19794 but also introduces a regression due to [removal of `ErrTruncated`](https://github.com/miekg/dns/issues/814) which causes the following FTBFS:

~~~~
src/github.com/docker/libnetwork/resolver.go:487:28: undefined: "github.com/miekg/dns".ErrTruncated
~~~~

Please consider upgrading `github.com/miekg/dns`. Thanks.

### PR DESCRIPTION
**Pull request by SamWhited**
Bumps the DNS library in use to the latest version so that Docker can eventually follow suit.

I'm not sure why there were extra `vendor/` changes, to vendor after bumping the commit I ran:

```
go get -u github.com/LK4D4/vndr@85886e1ac99b8d96590e6e0d9f075dc7a711d132
vndr
```

Fixes #2493

### CODE REVIEW

**Comment by thaJeztah** [comment]
> I'm not sure why there were extra vendor/ changes

could be if last vendoring was done with an older version of vndr, or go?

what I usually do in that case, is to first do a full vendor cycle with `vndr` to check if it's unrelated; commit those changes (if any), using a commit message like "re-vendor with latest vndr", then update the vendor.conf, and vendor the updated dependency

**Comment by SamWhited** [comment]
I used the version of vndr from moby/moby's dockerfile assuming that would be correct; I'll split it out into two commits.

**Comment by SamWhited** [comment]
Done

**Comment by thaJeztah** [comment]
full diff of the changes in the dependency; https://github.com/miekg/dns/compare/v1.0.7...v1.1.27

File: resolver.go [hunk 1]
```diff
@@ -484,7 +484,7 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 			resp, err = co.ReadMsg()
 			// Truncated DNS replies should be sent to the client so that the
 			// client can retry over TCP
-			if err != nil && err != dns.ErrTruncated {
+			if err != nil && !resp.Truncated {
```
Comment (thaJeztah) [inline resolver.go]: This Error seems to have been removed in https://github.com/miekg/dns/pull/815

I ❤️ SemVer

**Comment by thaJeztah** [comment]
Looks like this closes https://github.com/docker/libnetwork/issues/2493 (could you edit the top comment to add that?)

**Comment by SamWhited** [comment]
> Looks like this closes #2493 (could you edit the top comment to add that?)

Done, thanks

**Comment by onlyjob** [comment]
Thanks!

**Comment by thaJeztah** [comment]
Looks like this change caused a regression; https://github.com/moby/moby/issues/40634#issuecomment-595824120

**Pull request merged**